File: face_comparsion/src/main/java/com/netposa/poseidon/face/init/ThreadExceptionHandler.scala
package com.netposa.poseidon.face.init

import org.slf4j.{Logger, LoggerFactory}


/**
  * 线程处理过程中的异常处理工具
  */
class ThreadExceptionHandler extends Thread.UncaughtExceptionHandler {
    val LOG: Logger = LoggerFactory.getLogger("ThreadExceptionHandler")
    override def uncaughtException(t: Thread, e: Throwable) {
        LOG.error("thread=>%d,error msg is=>%s".format(t.getId, e.getMessage), e)
        val newThread = new Thread(t, t.getName)
        newThread.setUncaughtExceptionHandler(this)
        newThread.start()
    }
}

File: fackcar/src/main/java/netposa/fakecar/feature/rpc/util/ThreadExceptionHandler.java
package netposa.fakecar.feature.rpc.util;

import java.lang.Thread.UncaughtExceptionHandler;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * 线程失败时的异常处理流程
 * @author hongs.yang
 *
 */
public class ThreadExceptionHandler implements UncaughtExceptionHandler {
	private static final Logger LOG = LoggerFactory.getLogger(ThreadExceptionHandler.class);

	@Override
	public void uncaughtException(Thread t, Throwable e) {
		LOG.warn("this thread " + t.getName() + ", execute error => " + e.getMessage(),e);
	}

}
